server/routes/alerting.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { Request, ResponseToolkit } from 'hapi';
+//@ts-ignore
+import { get, set } from 'lodash';
+//@ts-ignore
+import { CallClusterWithRequest } from 'src/legacy/core_plugins/elasticsearch';
+import { SearchResponse } from '../models/interfaces';
+import { Monitor, ServerResponse } from '../models/types';
+import { Router } from '../router';
+
+export default function(apiRouter: Router) {
+  apiRouter.post('/monitors/_search', searchMonitors);
+}
+
+const searchMonitors = async (
+  req: Request,
+  h: ResponseToolkit,
+  callWithRequest: CallClusterWithRequest
+): Promise<ServerResponse<any>> => {
+  try {
+    const requestBody = {
+      query: {
+        nested: {
+          path: 'monitor.inputs',
+          query: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    'monitor.inputs.search.indices.keyword': {
+                      value: '.opendistro-anomaly-results*',
```

Comment:
Why search value is index name ?

Reply:
For AD monitors, we will use AD result index and we can search AD monitors with this.

Reply:
One sample for monitor inputs
```
...
"inputs" : [
            {
              "search" : {
                "indices" : [
                  ".opendistro-anomaly-results*"
                ],
                "query" : {
                  "size" : 1,
                  "query" : {
                    "bool" : {
                      "filter" : [
                        {
                          "range" : {
                            "data_end_time" : {
                              "from" : "{{period_end}}||-2m",
                              "to" : "{{period_end}}",
                              "include_lower" : true,
                              "include_upper" : true,
                              "boost" : 1.0
                            }
                          }
                        },
                        {
                          "term" : {
                            "detector_id" : {
                              "value" : "LNULDnEBdsd5fFw6Wr6N",
                              "boost" : 1.0
                            }
                          }
                        }
                      ],
                      "adjust_pure_negative" : true,
                      "boost" : 1.0
                    }
                  },
                  "sort" : [
                    {
                      "anomaly_grade" : {
                        "order" : "desc"
                      }
                    },
                    {
                      "confidence" : {
                        "order" : "desc"
                      }
                    }
                  ],
                  "aggregations" : {
                    "max_anomaly_grade" : {
                      "max" : {
                        "field" : "anomaly_grade"
                      }
                    }
                  }
                }
              }
            }
          ]
...
```